vspace/src/main/java/edu/asu/diging/vspace/core/factory/impl/ChoiceBlockFactory.java

```diff
@@ -0,0 +1,33 @@
+package edu.asu.diging.vspace.core.factory.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.factory.IChoiceBlockFactory;
+import edu.asu.diging.vspace.core.model.IChoice;
+import edu.asu.diging.vspace.core.model.IChoiceBlock;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.ChoiceBlock;
+
+@Service
+public class ChoiceBlockFactory implements IChoiceBlockFactory {
+    
+    /* (non-Javadoc)
+     * @see edu.asu.diging.vspace.core.factory.impl.IChoiceBlockFactory#createChoiceBlock(edu.asu.diging.vspace.core.model.impl.ISlide, java.lang.Integer,
+     * edu.asu.diging.vspace.core.model.impl.IChoice)
+     */    
+    @Override
+    public IChoiceBlock createChoiceBlock(ISlide slide, Integer contentOrder, List<IChoice> choices) {
+        IChoiceBlock choiceBlock = new ChoiceBlock();
+        List<IChoice> choicesList= new ArrayList<IChoice>();
+        for(int i=0;i<choices.size();i++) {
+            choicesList.add(choices.get(i));
+        }
+        choiceBlock.setSlide(slide);
+        choiceBlock.setContentOrder(contentOrder);
+        choiceBlock.setChoices(choicesList);
```

Comment:
I wouldn't set the choices here. since it should show all choices of a branching point, the block would have to be updated when new choices are added, existing one deleted, etc.

Reply:
So, add choices option on staff slide screen should allow user to add/remove choices to slide?

Reply:
the user should be able to add choice blocks. And a choice block would for now always show all choices. The user could add two or more choice blocks (e.g. beginning and end of slide) and they would just show the same choices.

Reply:
but there need to be choice blocks, as in a next step we would allow the user to select a layout and the layout would specify where what blocks go.